### DISCUSSION TITLE
How to keep the gradient when hovering over an element?

### DISCUSSION THREAD

**Discussion by wiredmatt** [mantinedev/mantine#1940] [Q&A] [answered]
Hi, I'd like for my button to not change the gradient I gave it when it gets hovered.

Component:

```tsx
....
<M.Button
  variant="gradient"
  gradient={{ from: '#4f10fe', to: '#a917e0', deg: 90 }}
  size="lg"
>
  A button
</M.Button>
...
```

I tried overriding styles even but it does nothing:

```tsx
<M.Button
  variant="gradient"
  gradient={{ from: '#4f10fe', to: '#a917e0', deg: 90 }}
  size="lg"
  color={'#4f10fe'}
  styles={theme => ({
    root: {
      ':hover': {
        backgroundColor: theme.fn.linearGradient(
          90,
          '#4f10fe',
          '#a917e0'
        )
      }
    }
  })}
>
  A button
</M.Button>
```

**Comment by nickngqs** [comment]
```js
styles={(theme) => ({
    root: {
        ':hover': {
                backgroundImage: theme.fn.linearGradient(90,'#4f10fe', '#a917e0'),
		':hover': { backgroundSize: '100%', },
        },
    },
})}
```

**Reply by wiredmatt** [reply]
That's completely unintuitive, but thanks! There should be a cleaner way to achieve this though.

**Reply by nickngqs** [reply]
the nested :hover adds an additional backgroundSize. Not sure why it does that though, not sure if its a bug or there is a reason for that.

while backgroundImage is a css property needed for gradient, you can't use gradient for backgroundColor.

**Comment by rtivital** [comment, marked as answer]
For 5.0+ `<Button sx={{ '&:hover': { backgroundImage: theme.fn.gradient({ from: '#4f10fe', to: '#a917e0', deg: 90 }) } }} />`

**Reply by wiredmatt** [reply]
Looks better, but would love to see a simple bool prop that would disable changing the background on hover instead. Thanks!
